src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +102,1361 @@ static DamageTable D_80BBF858 = {
+                return;
+            }
+            this->textID = EnSGoro_BombshopGoron_NextTextID(this, play);
+            Message_StartTextbox(play, this->textID, &this->actor);
+        }
+    } else if ((msgState == 4) && (Message_ShouldAdvance(play))) {
+        switch (play->msgCtx.choiceIndex) {
+            case 0:
+                this->bombbuyFlags |= ENSGORO_BOMBBUYFLAG_YESBUY;
+                break;
+            case 1:
+                func_8019F230();
+                this->bombbuyFlags &= ~ENSGORO_BOMBBUYFLAG_YESBUY;
+                break;
+        }
+
+        this->textID = EnSGoro_BombshopGoron_NextTextID(this, play);
+        if ((this->textID == 0x675) || (this->textID == 0x676)) {
+            play->msgCtx.msgMode = 0x43;
+            play->msgCtx.unk12023 = 4;
+            Actor_PickUp(&this->actor, play, GI_POWDER_KEG, 300.0f, 300.0f);
+            this->actionFunc = EnSGoro_ActionShopTakePayment;
+        } else {
+            Message_StartTextbox(play, this->textID, &this->actor);
+        }
+    }
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000, 0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+}
+
+void EnSGoro_ActionShopTakePayment(EnSGoro* this, PlayState* play) {
+    if (Actor_HasParent(&this->actor, play)) {
+        this->actor.parent = NULL;
+        func_801159EC(this->powderKegPrice * -1);
+        this->actionFunc = EnSGoro_ActionShopFinishTransaction;
+    } else {
+        Actor_PickUp(&this->actor, play, GI_POWDER_KEG, 300.0f, 300.0f);
+    }
+}
+
+void EnSGoro_ActionShopFinishTransaction(EnSGoro* this, PlayState* play) {
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        Message_StartTextbox(play, this->textID, &this->actor);
+        this->actionFunc = EnSGoro_ActionShopGoronTalking;
+        return;
+    }
+    func_800B85E0(&this->actor, play, 400.0f, -1);
+}
+
+void EnSGoro_ActionAsleep(EnSGoro* this, PlayState* play) {
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        Message_StartTextbox(play, 0x23A, &this->actor);
+        this->actionFunc = EnSGoro_ActionAsleepMessageWait;
+    } else if (this->actor.isTargeted) {
+        func_800B863C(&this->actor, play);
+    }
+    EnSGoro_UpdateSleeping(this, play);
+}
+
+void EnSGoro_ActionAsleepMessageWait(EnSGoro* this, PlayState* play) {
+    if (Message_GetState(&play->msgCtx) == 6) {
+        if (Message_ShouldAdvance(play)) {
+            play->msgCtx.msgMode = 0x43;
+            play->msgCtx.unk12023 = 4;
+            this->actionFunc = EnSGoro_ActionAsleep;
+        }
+    }
+    EnSGoro_UpdateSleeping(this, play);
+}
 
-extern ColliderCylinderInit D_80BBF820;
-extern CollisionCheckInfoInit2 D_80BBF84C;
-extern DamageTable D_80BBF858;
+void EnSGoro_Init(Actor* thisx, PlayState* play) {
 
-extern UNK_TYPE D_060091A8;
-extern UNK_TYPE D_06012DE0;
+    EnSGoro* this = THIS;
+    SkelAnime* skelAnime = &this->skelAnime;
```

Comment:
Is this skelAnime temp really needed? Could you just use `&this->skelAnime` directly, and then just have an `s32 pad` above `this`?